fairease/udal/brokers/beacon.py

```diff
@@ -0,0 +1,177 @@
+import datetime
+import json
+from pathlib import Path
+import tempfile
+from typing import List
+import requests
+import os
+from dotenv import load_dotenv
+import xarray as xr
+
+
+
+from ..broker import Broker
+from ..namedqueries import NamedQueryInfo, QueryName, QUERY_NAMES, QUERY_REGISTRY
+from ..result import Result
+from ..config import Config
+
+beaconBrokerQueryName: List[QueryName] = [
+    'urn:fairease.eu:argo'
+]
+
+beaconBrokerQueries: dict[QueryName, NamedQueryInfo] = \
+    { k: v for k, v in QUERY_REGISTRY.items() if k in beaconBrokerQueryName }
+
+
+class BeaconBroker(Broker):
+
+    _config: Config
+
+    _queryNames: List[QueryName] = beaconBrokerQueryName
+
+    _queries: dict[QueryName, NamedQueryInfo] = beaconBrokerQueries
+
+    @property
+    def queryNames(self) -> List[str]:
+        return list(BeaconBroker._queryNames)
+
+    @property
+    def queries(self) -> List[NamedQueryInfo]:
+        return list(BeaconBroker._queries.values())
+
+    def __init__(self, config: Config):
+        
+        self._config = config
+        if not self._config or not self._config.beacon_token:
+            raise ValueError('Please provide a token')
```

Comment:
```suggestion
            raise Exception('Please provide a beacon token in the UDAL configuration')
```
I do not think a `ValueError` is the right exception, since no value is provided (could also be a `TypeError` since we would not know the type of the value). Since there is no built-in exception type for this case, I would just use `Exception`.